graphql_compiler/compiler/metadata.py

```diff
@@ -47,6 +54,21 @@ def register_location(self, location, location_info):
             raise AssertionError(u'Attempting to register an already-registered location {}: '
                                  u'old info {}, new info {}'
                                  .format(location, old_info, location_info))
+
+        if location.field is not None:
+            raise AssertionError(u'Attempting to register a location at a field, this is '
+                                 u'not allowed: {} {}'.format(location, location_info))
+
+        if location_info.parent_location is None:
+            # Only the root location and revisits of the root location are allowed
+            # to not have a parent location.
+            is_root_location = location == self._root_location
```

Comment:
I believe that one of these can be done away with if you wanted, since the root location will be a revisit of itself.

Reply:
I'm worried about confusion here, since "re"-visit seems to imply a time dependence as @lodrion and @amartyashankha point out, and I think it has to be specifically a visit different than the first. I'd prefer to make locations not revisits of themselves for that reason.